### DISCUSSION TITLE
Out-of-the box visualiser does not compile in the VS 2019.

### DISCUSSION THREAD

**Discussion by GeneralGDA** [artem-ogre/CDT#33] [General]
I took the whole CDT repo, ran CMAKE, opened VS solution and the code did not compile:
1) CDTUtils.hpp uses std::runtime_error, but there is no #include <stdexcept> in it.
2) main.cpp, line 324: pts.begin() should be replaces with pts.data.

**Comment by artem-ogre** [comment]
Спасибо за информацию.

Ты на master'е?
Как выглядит команда вызова cmake'а?

2. Эта строчка (322)? https://github.com/artem-ogre/CDT/blob/master/visualizer/main.cpp#L322 
  Как выглядит ошибка?

**Reply by artem-ogre** [reply]
Я не спорю, что это ошибки, но мне просто интересно почему гитхабовские билды проходят. Например здесь: https://github.com/artem-ogre/CDT/runs/1824309079?check_suite_focus=true

**Reply by GeneralGDA** [reply]
0) Да, я текущий мастер скачал.
1) Команда cmake, стоя в папке visualizer: cmake ./ 
2) Да. error C2664: 'void QPainter::drawPolygon(const QPointF *,int,Qt::FillRule)': cannot convert argument 1 from 'std::_Array_const_iterator<_Ty,3>' to 'const QPointF *'

**Reply by artem-ogre** [reply]
Спасибо ещё раз! Пушанул фиксы в мастер. Починилось?

**Reply by GeneralGDA** [reply]
Да.

**Comment by artem-ogre** [comment]
Fixed in: https://github.com/artem-ogre/CDT/commit/a98f17ff412aa7cc82924df09e8246b02e49d92c
